rholang/src/main/scala/coop/rchain/rholang/interpreter/Registry.scala

```diff
@@ -572,6 +628,73 @@ class Registry(private val space: ISpace[Channel,
       case _ => Task.unit
     }
 
+  def publicLookup(args: RootSeq[ListChannelWithRandom]): Task[Unit] =
+    args match {
+      case Seq(ListChannelWithRandom(Seq(key, ret), rand, cost)) =>
+        def localFail() = fail(ret, rand)
+        try {
+          val Channel(Quote(keyPar)) = key
+          val Some(Expr(GUri(uri)))  = keyPar.singleExpr
+          if (uri.startsWith("rho:id:")) {
+            val tail = uri.substring(7)
+            if (tail.size != 54) {
+              localFail()
+            } else {
+              // Could fail
+              // 256 bits plus 14 bit crc-14
+              val bytes: Array[Byte] = ZBase32.decode(tail, 270)
+              val crc: Short         = ((bytes(32).toShort & 0xff) | ((bytes(33).toShort & 0xfc) << 6)).toShort
+              if (crc == CRC14.compute(bytes.view.slice(0, 32))) {
```

Comment:
Can you explain why do we need CRC?

Reply:
These are intended for transcription by hand, hence the base32 encoding.

We want to be able to tell a user if their transcription was invalid.
We can detect a mistake in transcription with probability (1 -  1/16000).
We use CRC14 because of the base32 encoding, and the choice of 256 bits. 256 bits is 52 base 32 characters, but the last character only stores 1 bit. We use the 4 leftovers and 2 more for 14. The CRC-14 is CRC-14-DARC

Reply:
Add that design rationale as a comment in the code, perhaps?